### PR TITLE
Pull Request for Issue1113: AMExporter File Names

### PR DESCRIPTION
**Pull request by davidChevrier**
Fix to make sure that exported file names don't have non-printable characters in them.

### CODE REVIEW

File: source/dataman/export/AMExporterGeneralAscii.cpp [hunk 1]
```diff
@@ -93,6 +93,20 @@ QString AMExporterGeneralAscii::exportScan(const AMScan *scan, const QString &de
 
 	// prepare export file
 	mainFileName_ = parseKeywordString( destinationFolderPath % "/" % option->fileName() );
+	mainFileName_ = removeNonPrintableCharacters(mainFileName_);
+
+//	bool detectedNonprintable = false;
```
Comment (dretrex) [inline source/dataman/export/AMExporterGeneralAscii.cpp]: Unnecessary comments.  Probably should remove these.

File: source/dataman/export/AMExporter.h [hunk 2]
```diff
@@ -200,6 +200,9 @@ public slots:
 
 	QString krFileSystemAutoIncrement(const QString &arg = QString());
 
+	/// Helper function to remove non-printable characters from a file name
+	QString removeNonPrintableCharacters(const QString &fileName);
```
Comment (dretrex) [inline source/dataman/export/AMExporter.h]: I think technically this should be a const function.

File: source/dataman/export/AMExporterGeneralAscii.cpp [hunk 3]
```diff
@@ -516,6 +530,19 @@ bool AMExporterGeneralAscii::writeSeparateFiles(const QString& destinationFolder
 
 			QFile file;
 			QString separateFileName = parseKeywordString( destinationFolderPath % "/" % option_->separateSectionFileName() );
+			separateFileName = removeNonPrintableCharacters(separateFileName);
+//			bool detectedNonprintable = false;
```
Comment (dretrex) [inline source/dataman/export/AMExporterGeneralAscii.cpp]: Commented code.

File: source/dataman/export/AMExporterGeneralAscii.cpp [hunk 4]
```diff
@@ -599,6 +626,22 @@ bool AMExporterGeneralAscii::writeSeparateFiles(const QString& destinationFolder
 			QFile file;
 			QString separateFileName = parseKeywordString( destinationFolderPath % "/" % option_->separateSectionFileName() );
 
+			bool detectedNonprintable = false;
```
Comment (dretrex) [inline source/dataman/export/AMExporterGeneralAscii.cpp]: Is there a reason why you're not just using the function you made here?  They look pretty similar to me.

Comment (davidChevrier) [inline source/dataman/export/AMExporterGeneralAscii.cpp]: Thanks @dretrex good catch. I found a couple places I missed because of this.

**Comment by davidChevrier** [comment]
Time Estimate: 8 hours
(with diagnostics and testing)

**Pull request merged**